_studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp

```diff
@@ -1050,7 +1050,8 @@ mfxStatus VideoDECODEH265::DecodeFrameCheck(mfxBitstream *bs, mfxFrameSurface1 *
             {
                 if (!m_isFirstRun)
                 {
-                    sts = MFX_WRN_VIDEO_PARAM_CHANGED;
```

Comment:
I'm not suggesting change this logic in VPL. MFX_WRN_VIDEO_PARAM_CHANGED means the decoder detected a new sequence header. Some parameters may be changed in the new sequence. For some parameters changed, decoder can still work fine without re-init decoder. Please try to call  MFXVideoDECODE_GetVideoParam(session, &newParams) to get the new parameters after MFXVideoDECODE_DecodeFrameAsync returned MFX_WRN_VIDEO_PARAM_CHANGED in Codec2.0, if color aspects really differ, then close the decoder and re-init decoder.

Reply:
Hi Tianmi, I agree with you. re-checking colorAspect and bitdepth in codec2.0 sounds good. Change it here is just for keeping HEVC behavior same with AVC. If re-checking parameters in codec2.0, we need to add a if branch for HEVC only.